### ISSUE TITLE
Inconsistent leading in footnotes.

### ISSUE THREAD

**Issue by alerque** [simoncozens/sile#213]
For some reason the first line (and first line only, not the first footnote insert) sometimes shows up with a much larger leader than its cousins. Note the extra leading where marked:

![selection_107](https://cloud.githubusercontent.com/assets/173595/11265536/94085106-8ea5-11e5-9d7b-d60e3870a9fc.png)

One is manifesting after the first line in a wrapped comment, the others ore in the inter-insert space. I haven't been able to find what triggers this, but when extra space is added it seems to be always after the first line. Note about 50% of pages are fine, but this is pretty frequent. This is possibly related to #212 but distinct in that it's never squished and always the first line.

My current footnote code (used in the example above) is in [this comment](https://github.com/simoncozens/sile/issues/211#issuecomment-157983989).

**Comment by simoncozens** [comment]
Can you see if this still happens with the latest insertion package? If so, I may need a MWE, unfortunately - I can't reproduce it.

**Comment by alerque** [comment]
I'm out of town right now but I'll re-run this as soon as I get a chance and report back where it's at.

**Comment by alerque** [comment]
I'm getting nothing but an error when typesetting any of my books that have footnotes. Others without seem to work fine.

    Error detected:
    /usr/local/share/sile/core/utilities.lua:111: attempt to get length of a nil value (local 't2')

**Comment by alerque** [comment]
So it's not actually failing on _all_ footnotes, it's actually throwing in the towel on the very ones that, in the previous version I was using, would have ended up overlapping the body text.

**Comment by alerque** [comment]
So with the patch above, this is still not fixed. The first line of footnotes is still not consistently spaced with. But it seems like it's not just the first line any more, sometimes the last one in affected.

![selection_112](https://cloud.githubusercontent.com/assets/173595/11429704/687437a0-9484-11e5-9554-c867389fd5cc.png)

**Comment by simoncozens** [comment]
As per slack, I think the issue is that leading in footnotes is applied according to the usual TeX rules, i.e. lines with descenders will have different spacing to lines without. Even if the body text has fill-font spacing, we reset the settings when we box up footnote content.

Setting inter-insertion-skip to zero shows the problem:

![screen shot 2015-11-27 at 16 09 42](https://cloud.githubusercontent.com/assets/106728/11435752/4bd16dbe-9521-11e5-99e4-1fef876b2a7d.png)

And that's even with

```TeX
\define[command=footnote:atstart]{\set[parameter=linespacing.method,value=fit-font]\set[parameter=linespacing.fit-font.extra-space,value=20pt]X}
```

But of course that won't work because letting is only applied *between* lines.

TLDR: Leading should be added between footnotes, and isn't being.

**Comment by simoncozens** [comment]
I'm getting confused. Leading is a distraction. The inter-insertion-skip *is* the leading between footnotes. The actual problem is that box depth is not being used in insertoin height calcuations.

**Comment by alerque** [comment]
This looks _sooooo_ much better. I owe you one @simoncozens.
